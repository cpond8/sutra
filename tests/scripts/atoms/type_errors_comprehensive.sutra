;; Comprehensive Type Error Testing
;; Tests that enhanced type error messages provide clear guidance
;; Each test should fail with specific type error and helpful suggestions

;; Arithmetic operations with wrong types
(+ "hello" 5)           ; string + number - should suggest conversion