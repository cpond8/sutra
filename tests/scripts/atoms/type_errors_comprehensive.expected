Evaluation error: SutraError { kind: Eval(EvalError { message: "Type mismatch in call to '+'\n\nExpected argument of type Number, but received String.\n\nArgument provided: string \"hello\" (from expression: string \"hello\")", expanded_code: "\"hello\"", original_code: None, suggestion: Some("Arithmetic operations require numbers. Convert strings to numbers if needed Consider parsing the string to a number if it contains numeric data.") }), span: Some(Span { start: 219, end: 232 }) }